code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -121,7 +122,13 @@
 
 /mob/living/silicon/robot/proc/is_component_functioning(module_name)
 	var/datum/robot_component/C = components[module_name]
-	return C && C.installed == 1 && C.toggled && C.is_powered()
+	return C && C.installed == 1 && C.toggled && C.is_powered() && !C.component_disabled
+
+/mob/living/silicon/robot/proc/disable_component(module_name, duration)
+	var/datum/robot_component/D = get_component(module_name)
+	D.component_disabled = 1
+	spawn(duration)
+		D.component_disabled = 0
```

Comment:
Spawn overlap will break this - if a borg gets severely EMP'd once, then severely EMP'd again 155 seconds later, they'll get their binary back 5 seconds later. Use `++` instead of `= 1` and `--` instead of `= 0` to make sure it doesn't get reset prematurely.